SU-SWS/stanford_paragraph_types
# Hsdo 932 field option perms

# READY FOR REVIEW

# Summary
- Created options permissions to limit the available options in blockrefence fields. This will limit the options for particular roles or disable the field if the field is pre-populated (edit form or something)

# Needed By (Date)
- End of sprint (8/3)

# Urgency
- Low

# Steps to Test

1. Checkout branch. ensure `stanford_paragraph_types_dynamic` is enabled
2. Use the dynamic paragraph type or create a block reference field on anything.
3. In the field settings, there should be a link at the top that says "Options Permissions" near the "widget Type" link
4. click on "Options Permissions" and configure the available options with the roles to associate with.
5. save and create a piece of content with that field as admin
6. verify the field options are not limited.
7. log in as a user with a role you configured
8. create/edit a piece of content with the field above
9. verify either the options are limited or that the field is disabled.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
